Services/UI/OJS.Services.Ui.Business/Implementations/ParticipantsBusinessService.cs

```diff
@@ -42,21 +42,20 @@ public class ParticipantsBusinessService : IParticipantsBusinessService
         {
             participant.ParticipationStartTime = DateTime.Now;
             participant.ParticipationEndTime = DateTime.Now + contest.Duration;
-
-            var isUserLecturerInByContestAndUser =
-                await this.contestsData.IsUserLecturerInByContestAndUser(contest.Id, userId);
-
-            if (!isAdmin && !isUserLecturerInByContestAndUser)
-            {
-                AssignRandomProblemsToParticipant(participant, contest);
-            }
         }
         else if (isOfficial && contest.IsOnsiteExam)
         {
             participant.ParticipationStartTime = DateTime.Now;
             participant.ParticipationEndTime = contest.EndTime;
         }
 
+        var isUserLecturerInByContestAndUser =
+            await this.contestsData.IsUserLecturerInByContestAndUser(contest.Id, userId);
+        if (!isAdmin && !isUserLecturerInByContestAndUser)
+        {
+            AssignRandomProblemsToParticipant(participant, contest);
+        }
+
```

Comment:
We should not assign random problems to users in Practice mode. Also will this work if the contest is not Online?